thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -249,4 +268,44 @@ public void run() {
 
     return Response.ok().build();
   }
+
+
+  @POST
+  @Path("/autotune/{id}")
+  public Response tuneAlertFilter(@PathParam("id") String id,
+      @QueryParam("startTime") Long startTime, @QueryParam("endTime") Long endTime)
+      throws Exception {
+    AnomalyFunctionDTO anomalyFunctionSpec = DAO_REGISTRY.getAnomalyFunctionDAO().findById(Long.valueOf(id));
+    AnomalyFunctionManager anomalyFunctionDAO = DAO_REGISTRY.getAnomalyFunctionDAO();
+    AlertFilter alertFilter;
+    Map<String, String> alertFilterParams = anomalyFunctionSpec.getAlertFilter();
+    if (alertFilterParams == null) {
+      alertFilter = new DummyAlertFilter();
+    } else if (alertFilterParams.get("type").toUpperCase().equals(ALPHA_BETA.toString())) {
+      alertFilter = new AlphaBetaAlertFilter();
+      alertFilter.setParameters(alertFilterParams);
+    } else {
+      alertFilter = alertFilterFactory.fromSpec(anomalyFunctionSpec);
+      alertFilter.setParameters(alertFilterParams);
+    }
+    MergedAnomalyResultManager anomalyMergedResultDAO = DAO_REGISTRY.getMergedAnomalyResultDAO();
+    List<MergedAnomalyResultDTO> anomalyResultDTOS = anomalyMergedResultDAO.findByStartTimeInRangeAndFunctionId(startTime,endTime,Long.valueOf(id));
+    AlertFilterUtil evaluator = new AlertFilterUtil(alertFilter);
+    try {
+      //evaluate current alert filter (calculate current precision and recall)
+      double[] evals = evaluator.getEvalResults(anomalyResultDTOS);
+      AlertFilterAutoTune alertFilterAutotune = alertFilterAutotuneFactory.fromSpec(anomalyFunctionSpec);
+      Map<String,String> tunedAlertFilter = alertFilterAutotune.tuneAlertFilter(anomalyResultDTOS, evals[AlertFilterUtil.PRECISION_INDEX], evals[AlertFilterUtil.RECALL_INDEX]);
+      if (alertFilterAutotune.isUpdated()){
+        anomalyFunctionSpec.setAlertFilter(tunedAlertFilter);
+        anomalyFunctionDAO.update(anomalyFunctionSpec);
+        alertFilter.setParameters(tunedAlertFilter);
+      } else {
+        LOG.info("Model hasn't been updated");
```

Comment:
Add more information for debugging purpose. For instance, we could log which function's alert filter is just run through the auto tune procedure.